### PR TITLE
Planthunt

### PR DESCRIPTION
**Pull request by Zephyr75**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java [hunk 1]
```diff
@@ -90,6 +105,7 @@ public void onClick(View v) {
             }
         });
 
+        System.out.println(hostStatus);
```
Comment (Aco-Hub) [inline app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java:108]: Delete the Sysouts in your code.

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java [hunk 2]
```diff
@@ -214,10 +255,27 @@ public void run() {
                                     try  {
                                         //Returns built URL with given image link
                                         String urlString = PlantnetApi.buildUrl(PlantnetApi.API_KEY, urlImage, "flower");
+                                        System.out.println(urlString);
```
Comment (Aco-Hub) [inline app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java:258]: delete Sysout

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java [hunk 3]
```diff
@@ -214,10 +255,27 @@ public void run() {
...
+                                            Snackbar.make(findViewById(R.id.planthuntLobbyLayout), "No plant was found", Snackbar.LENGTH_LONG).show();
+                                            storage.child("Planthunt").child(user.getUid()).child(photoFile.getName()).delete();
+                                            return;
+                                        }
+
+                                        //Gets how common the plant is
+                                        int popularity = 0;
+                                        for (int i = 0; i < json.getJSONArray("results").length(); i++){
+                                            if (Double.parseDouble(json.getJSONArray("results").getJSONObject(i).get("score").toString()) > .05){
+                                                popularity++;
```
Comment (Aco-Hub) [inline app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java:275]: Good system of popularity.

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntMainActivity.java [hunk 4]
```diff
@@ -47,9 +65,30 @@ public void onClick(View v) {
...
     }
+
+    public boolean internetIsConnected() {
+        try {
+            String command = "ping -c 1 google.com";
+            return (Runtime.getRuntime().exec(command).waitFor() == 0);
+        } catch (Exception e) {
+            return false;
+        }
+    }
```
Comment (Aco-Hub) [inline app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntMainActivity.java:93]: Creative code to check if you are connected.

**Review by Aco-Hub — COMMENTED** [review]
Just some parts of the code to cleanup

**Review by Aco-Hub — CHANGES_REQUESTED** [review]
Just some part of the code to cleanup

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java (hunk 3 above)
Comment (Zephyr75) [reply app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivity.java:275]: Thanks!

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntMainActivity.java (hunk 4 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntMainActivity.java:93]: +1

**Pull request merged**